testkit/gradle-testkit-truth/src/main/kotlin/com/autonomousapps/kit/truth/BuildResultSubject.kt

```diff
@@ -37,11 +38,25 @@ public class BuildResultSubject private constructor(
     return check("getOutput()").that(actual!!.output)
   }
 
-  public fun task(name: String): BuildTaskSubject {
+  @CanIgnoreReturnValue
+  public fun task(path: String): BuildTaskSubject {
     if (actual == null) {
       failWithActual(simpleFact("build result was null"))
     }
-    return check("task(%s)", name).about(buildTasks()).that(actual!!.task(name))
+    val tasks = actual!!.tasks.map { it.path }
+
+    check("getTasks()").that(tasks).contains(path)
+
+    return check("task(%s)", path).about(buildTasks()).that(actual.task(path))
+  }
+
+  public fun hasNotTask(path: String) {
```

Comment:
I'm not too happy with this function name. Any idea is more than welcome

Reply:
`checkHasTask`?

Reply:
Is there anything in the core Truth library that is similar that we can use for inspiration? I'm not too happy with it either, but I could live with it....

Reply:
`doesNotContainTask`? Similar to `doesNotContain`. The method that I use inside this function.

Reply:
how about `doesNotHaveTask`, since it uses the same verb as `hasTask`? you could also add `doesNotContainTask` since people are probably used to the verb "contain", and it could do the same thing (one function simply calling the other).

Reply:
Renamed to `doesNotHaveTask`. I think that, for now, one method is enough.

The thing I'm not sure about is this:

> since it uses the same verb as `hasTask`

There is not a `hasTask` function right now. Should I rename `task` to `hasTask`?

- `task` is more aligned with the gradle api.
- `hasTask` is more aligned with the truth api and with the fact that we have `doesNotHaveTask`.

Reply:
Oh huh, I'm not sure why I thought there was. Anyway, this is good enough. Thanks!